build/lib/src/builder/build_step_impl.dart

```diff
@@ -80,7 +80,8 @@ class BuildStepImpl implements BuildStep {
   @override
   Future writeAsBytes(AssetId id, FutureOr<List<int>> bytes) {
     _checkOutput(id);
-    var done = _futureOrWrite(bytes, (b) => _writer.writeAsBytes(id, b));
+    var done =
+        _futureOrWrite<List<int>>(bytes, (b) => _writer.writeAsBytes(id, b));
```

Comment:
I think I prefer it on the function literal, but I guess I would hope that it could be inferred by the type of bytes, which is known.

Since the type of bytes is T, and the type of the first arg is T, it should only need one or the other to infer both right?